include/picolibrary/stream.h

```diff
@@ -438,6 +440,78 @@ class Stream {
     Stream_Buffer * m_buffer{};
 };
 
+class Output_Stream;
+
+/**
+ * \brief Output formatter.
+ *
+ * \attention This class must be fully or partially specialized for each type that will be
+ *            printed.
+ *
+ * \tparam T The type to be printed.
+ */
+template<typename T, typename = void>
+class Output_Formatter {
```

Comment:
Move this class template to a dedicated header and remove the default implementation to avoid circular includes when specializing for types such as `::picolibrary::Error_Code`.

Reply:
Making this change would complicate documentation, and further header restructuring would be required.